tasks/localizationtasks.js

```diff
@@ -149,12 +148,62 @@ gulp.task('ext:localization:xliff-to-ts', function () {
 		.pipe(gulp.dest(config.paths.project.root + '/src/constants/'));
 });
 
+// Generates a package.nls.json file from localizedPackage.json.enu.xlf
+gulp.task('ext:localization:generate-eng-package.nls', function () {
+	return gulp.src([config.paths.project.localization + '/xliff/enu/localizedPackage.json.enu.xlf'], { base: '.' })
+		.pipe(through.obj(function (file, enc, callback) {
+			// convert xliff into json document
+			let dict = convertXmlToDictionary(String(file.contents), false);
+
+			var contents = ['{'];
+			var regxForReplacingQuots = new RegExp('"', 'g');
+
+			// Get all keys and values from localizedPackage.json.enu.xlf
+			Object.keys(dict).forEach(key => {
+				let value = '';
+				if (contents.length >= 2) {
+					contents[contents.length - 1] += ',';
+				}
+				if (dict.hasOwnProperty(key)) {
+					value = dict[key]['source'];
+				}
+				if (value && value.indexOf('"') >= 0) {
+					value = value.replace(regxForReplacingQuots, '\'');
+				}
+				let instantiation = '"' + key + '":"' + value + '"';
+				contents.push(instantiation);
+			});
+
+			// end the function
+			contents.push('}');
+
+			// Join with new lines in between
+			let fullFileContents = contents.join('\r\n') + '\r\n';
+			file.contents = new Buffer(fullFileContents);
+
+			let indexToStart = 'localizedPackage.json.'.length + 1;
+			let languageIndex = file.basename.indexOf('.', indexToStart);
+			let language = file.basename.substr(indexToStart - 1, (languageIndex - indexToStart) + 1);
+
+			file.basename = 'package.nls.json';
+
+			// Make the new file create on root
+			file.dirname = file.dirname.replace(language, '').replace('xliff', '').replace('localization', '');
+
+			// callback to notify we have completed the current file
+			callback(null, file);
+		}))
+		.pipe(gulp.dest(config.paths.project.root));
+})
+
 // Generates a localized package.nls.*.json
 gulp.task('ext:localization:xliff-to-package.nls', function () {
-	return gulp.src([config.paths.project.localization + '/xliff/**/localizedPackage.json.*.xlf', '!' + config.paths.project.localization + '/xliff/enu/localizedPackage.json.enu.xlf'], { base: '' })
+	return gulp.src([config.paths.project.localization + '/xliff/**/localizedPackage.json.*.xlf', '!' + config.paths.project.localization + '/xliff/enu/localizedPackage.json.enu.xlf'], { base: '.' })
 		.pipe(through.obj(function (file, enc, callback) {
 			// convert xliff into json document
 			let dict = convertXmlToDictionary(String(file.contents), false);
+			// Get the latest changes from package.nls.json after being updated.
+			var packageAllKeys = require('./../package.nls.json')
```

Comment:
Yeah, was going to point this out as the likely culprit here. I assume it's working better now? Everything else looks fine for what it's doing.

Reply:
Yep, it's working better